File: core/model/src/main/java/app/muplay/model/LibraryRole.java
package app.muplay.model;

/**
 * What a Navidrome library contains.
 *
 * <p>Navidrome hardcodes {@code child.Type = "music"} for every media file and
 * never signals that something is an audiobook, so this is a user decision made
 * once at setup — and it is the only mechanism available.
 */
public enum LibraryRole {
  MUSIC,
  AUDIOBOOKS,
  UNASSIGNED
}

File: core/model/src/main/kotlin/app/muplay/model/LibraryRole.kt
package app.muplay.model

/**
 * The role MuPlay assigns to a Subsonic music folder (what Navidrome calls a "library").
 *
 * The Subsonic `getMusicFolders` command returns only an `id` and an optional `name` for each
 * folder — nothing in the protocol response says what *kind* of content a folder holds. Every
 * [MusicLibrary] built directly from that response therefore carries [UNASSIGNED]; assigning
 * [MUSIC] or [AUDIOBOOKS] to a specific folder is necessarily a later, out-of-band decision (e.g.
 * matching a folder's name against what a setup flow expects) that this response shape alone
 * cannot make.
 */
enum class LibraryRole {
  MUSIC,
  AUDIOBOOKS,
  UNASSIGNED,
}
